src/jit/morph.cpp

```diff
@@ -4150,9 +4150,9 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
             // They should not affect the placement of any other args or stack space required.
             // Example: on AMD64 R10 and R11 are used for indirect VSD (generic interface) and cookie calls.
             isNonStandard = nonStandardArgs.FindReg(argx, &nonStdRegNum);
-            if (isNonStandard && (nonStdRegNum == REG_STK))
+            if (isNonStandard)
             {
-                isRegArg = false;
+                isRegArg = (nonStdRegNum != REG_STK);
             }
 #if defined(_TARGET_X86_)
             else if (call->IsTailCallViaHelper())
```

Comment:
Looks like this change makes this condition dead. It affects only `nonStandartArguments` and now they are all processed in the previous condition.

Reply:
Now the `else if (call->IsTailCallViaHelper())` condition is only executed if `!isNonStandard`. Previously, it was executed for `!isNonStandard || (nonStdRegNum != REG_STK))`. I think this is ok, because the `call->IsTailCallViaHelper()` case doesn't put anything in the nonStandardArgs list, so `!isNonStandard` will always be true. You could argue it would be more clear if we just remove the `else` altogether. I guess before, these two clauses could only make isRegArg false, so there was no need to check both conditions to do that. This case is simply another case of "non-standard args" that isn't handled by the "nonStandardArgs" mechanism. It's possible we could use the nonStandardArgs mechanism here, too, by tagging the last 4 arguments of `call->IsTailCallViaHelper()` calls as REG_STK. I'm not sure why we didn't do that; maybe because nonStandardArgs couldn't handle that at the time.